### PR TITLE
Add errorcode for null user in ATS

### PR DESCRIPTION
**Pull request by unpluggedk**
Addresses #209

### CODE REVIEW

**Comment by oldalton** [comment]
I'm not sure if we should also now update our guidance towards handling the new null user error. Current guidance is to show UI only in case of InteractionRequired error, but #209 kinda suggests that null user should be also handled with showing UI. @brandwe?

File: MSAL/src/public/MSALError.h [hunk 1]
```diff
@@ -94,10 +94,11 @@ typedef NS_ENUM(NSInteger, MSALErrorCode)
...
      */
-    MSALErrorInteractionRequired    = -42100,
-    MSALErrorMismatchedUser = -42101,
-    MSALErrorNoAuthorizationResponse = -42102,
-    MSALErrorBadAuthorizationResponse = -42103,
+    MSALErrorInteractionRequired        = -42100,
+    MSALErrorMismatchedUser             = -42101,
+    MSALErrorNoAuthorizationResponse    = -42102,
+    MSALErrorBadAuthorizationResponse   = -42103,
+    MSALErrorNullUser                   = -42104,
```
Comment (RPangrle) [inline MSAL/src/public/MSALError.h]: I think "MSALErrorUserRequired" is a bit better then "NullUser"

Comment (brandwe) [inline MSAL/src/public/MSALError.h]: I can update our guidance once the Error Code is agreed upon, but I think is exposing an issue with our model. _InteractionRequired is a control variable disguised as an Error Code._ I can imagine we'll have many Error conditions that could result in InteractionRequired as the resolution.

 Is there any way we can abstract out InteractionRequired or stack it?

Comment (unpluggedk) [inline MSAL/src/public/MSALError.h]: @brandwe I think this should be tracked as a separate issue.

**Pull request merged**